src/cocoa_init.m

```diff
@@ -359,8 +359,7 @@ void _glfwPlatformTerminate(void)
     _glfwTerminateNSGL();
     _glfwTerminateJoysticksNS();
 
-    [_glfw.ns.autoreleasePool release];
-    _glfw.ns.autoreleasePool = nil;
+    _glfwDestroyAutorealeasePool();
```

Comment:
I wonder if this was not a bug in the existing implementation : the doc says 

(https://developer.apple.com/documentation/foundation/nsautoreleasepool#1651537?language=objc)

> In a garbage-collected environment, sending a drain message to a pool triggers garbage collection if necessary; release, however, is a no-op.

Hence we should probably do a `drain` before the `release`, else some objects will not be properly destroyed?

Reply:
If you agree with my analysis but don'T want to change the behaviour (to have a conservative approach), you can drop a comment somewhere, maybe in the _glfwDestroyAutorealeasePool 
 function saying that it could be changed...